src/serialize/numberToHex.test.ts

```diff
@@ -4,4 +4,14 @@ describe("numberToHex", () => {
   it("can convert number to hex string", () => {
     expect(numberToHex(16)).toEqual("0x10");
   });
+  it("can pad", () => {
+    const str = "foo";
+    const pad = 10;
+    expect(numberToHex(16, { padStart: pad })).toEqual("0x0000000010");
```

Comment:
IMO a control value (`toEqual`) with a clear derivation would be more readable, like `0x` + `0`.repeat(pad) + `10`.
 ... and using a test value (pad=10) different than an output component (16 hex=> __10__) would be less confusing.